inductiva/mixins/file_manager.py

```diff
@@ -0,0 +1,230 @@
+        if source_file.endswith(TEMPLATE_EXTENSION):
+            if target_file is None:
+                new_target_file = new_target_file.split(TEMPLATE_EXTENSION)[0]
+
+            render_file(source_file=source_file,
+                        target_file=new_target_file,
+                        **render_args)
+        else:
+            if render_args:
+                logging.debug(
+                    f"Ignoring render_args since %s"
+                    f"isn't a {TEMPLATE_EXTENSION} file", source_file)
+            shutil.copy(source_file, new_target_file)
+
+        return new_target_file
+
+    def add_dir(self, source_dir, target_dir=None, **render_args):
+        """Render a directory from a template.
+        
+        Create a new directory inside the root_dir, keeping the same
+        structure as the source_dir, and replacing the render_args in
+        the template files. Only file with the template extension are replaced.
+        Other files are added as is.
+
+        Args:
+            source_dir: Path to the source directory.
+            target_dir: Path to the target directory.
+            render_args: Arguments to render the template files.
+        """
+        self.__check_root_dir()
+
+        if target_dir is None:
+            target_dir = "."
+
+        target_dir = os.path.join(self.__root_dir, target_dir)
+        shutil.copytree(source_dir,
+                        target_dir,
+                        dirs_exist_ok=True,
+                        symlinks=True)
+
+        if render_args:
+            template_files = get_template_files(target_dir)
+
+            for template_file in template_files:
+                template_path = os.path.join(target_dir, template_file)
+                target_path = template_path.split(TEMPLATE_EXTENSION)[0]
+
+                render_file(source_file=template_path,
+                            target_file=target_path,
+                            **render_args)
+                os.remove(template_path)
+
+        return target_dir
+
+
+def render_file(source_file, target_file: Union[str, io.StringIO],
+                **render_args):
+    """Render a file from a template.
+
+    Args:
+        source_file: Path to the source file.
+        target_file: Path to the target file.
+        render_args: Arguments to render the template file.
+    """
+
+    source_path = pathlib.Path(source_file)
+
+    source_dir = source_path.parent
+    source_file = source_path.name
+
+    environment = jinja2.Environment(loader=jinja2.FileSystemLoader(source_dir))
+    template = environment.get_template(source_file)
+    stream = template.stream(**render_args)
+    stream.dump(target_file)
+
+
+def get_template_files(src_dir):
+    """Get all template files in a directory."""
+
+    template_paths = glob.glob(os.path.join(src_dir, "**",
+                                            "*" + TEMPLATE_EXTENSION),
+                               recursive=True)
```

Comment:
```suggestion
    template_paths = glob.glob(
        os.path.join(src_dir, "**", "*" + TEMPLATE_EXTENSION),
        recursive=True)
```

Reply:
No luck! We have to change yapf, to avoid this problems. But that would move away from google style, so a more "general" discussion of code formatting is required so that every repo is uniform.